swap_meet/vendor.py

```diff
@@ -1,2 +1,139 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    """
+    Represents a vendor with a list of inventory items
+    """
+    def __init__(self,inventory = None):
+        """
+        instatiates Vendor instances; has attribute 'inventory', which
+        initialized as an empty list if no value is passed in.
+        """
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self,item):
+        """
+        Adds item to a vendor's inventory list
+        """
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        """
+        Removes an item from a vendor's inventory list if the item is
+        present and returns the item; otherwise returns False
+        """
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_id(self, id):
+        """
+        Returns an item by ID number; returns None if no item matches
+        the provided ID number
+        """
+        for item in self.inventory:
+            if id is item.id:
+                return item
+
+        return None
+
+    def swap_items(self, other_Vendor, my_item, their_item):
+        """
+        Swaps given item from one vendor's inventory list with an item
+        from another vendor's inventory list and returns True. If
+        either list is missing the given item, method returns False
+        """
+        if my_item in self.inventory and their_item in other_Vendor.inventory:
+            self.inventory.remove(my_item)
+            other_Vendor.inventory.append(my_item)
+
+            other_Vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, other_Vendor):
+        """
+        Swaps the first item in two vendor's inventory lists and
+        returns True; if either inventory list is empty, method returns
+        False
+        """
+        if self.inventory == [] or other_Vendor.inventory == []:
+            return False
+
+        else:
+            first_item_vendor = self.inventory[0]
+            first_item_other_vendor = other_Vendor.inventory[0]
+
+            if first_item_vendor and first_item_other_vendor:
+                self.inventory.remove(first_item_vendor)
+                other_Vendor.inventory.append(first_item_vendor)
+
+                other_Vendor.inventory.remove(first_item_other_vendor)
+                self.inventory.append(first_item_other_vendor)
+
+                return True
```

Comment:
One interesting point to note is that the README never specifies that the items need to be added to the end of each list! The tests don't check for a specific location for each item either. With this in mind, we know we are grabbing the first item from each list, so we could make this a true swap! One option would be to just reassign `self.inventory[0]` to `first_item_other_vendor` and `other_Vendor.inventory[0] to `first_item_vendor`.

We could also use a Python trick called multiple assignment. This allows us to assign multiple variables to new values all in one line. This might look like:

```suggestion
                self.inventory[0], other_Vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]

                return True
```

This technically creates a tuple out of the list retrievals on the right side of the equal sign and then unpacks the values into the two variables on the left! This can be a neat trick to assign multiple variables at once!